transport/tchannel/transport.go

```diff
@@ -333,22 +338,44 @@ func (t *Transport) onPeerStatusChanged(tp *tchannel.Peer) {
 // CreateTLSOutboundChannel creates a outbound channel for managing tls
 // connections with the given tls config and destination name.
 // Usage:
-// 	tr, _ := tchannel.NewTransport(...)
-//  outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
-//  outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
+//
+//		tr, _ := tchannel.NewTransport(...)
+//	 outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
+//	 outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
 func (t *Transport) CreateTLSOutboundChannel(tlsConfig *tls.Config, destinationName string) (peer.Transport, error) {
+	dialerFunc := t.dialer
+	if t.enableMPTCP {
+		dialerFunc = dialMPTCPContext
+	}
 	params := dialer.Params{
 		Config:        tlsConfig,
 		Meter:         t.meter,
 		Logger:        t.logger,
 		ServiceName:   t.name,
 		TransportName: TransportName,
 		Dest:          destinationName,
-		Dialer:        t.dialer,
+		Dialer:        dialerFunc,
 	}
 	return t.createOutboundChannel(dialer.NewTLSDialer(params).DialContext)
 }
 
+// CreateMPTCPOutboundChannel creates a outbound channel for MPTCP
+// connections
+// Usage:
+//
+//		tr, _ := tchannel.NewTransport(...)
+//	 outboundCh, _ := tr.CreateMPTCPOutboundChannel()
+//	 outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
+func (t *Transport) CreateMPTCPOutboundChannel() (peer.Transport, error) {
+	return t.createOutboundChannel(dialMPTCPContext)
```

Comment:
We should add warning log to let user know we are ignoring their dialer - if the dialer is passed in transport options